### PR TITLE
ENT-586: Add standalone setup.py for python-rhsm

### PR DESCRIPTION
**Pull request by vritant**

### CODE REVIEW

File: packages/rhsm/setup.cfg [hunk 1]
```diff
@@ -0,0 +1,4 @@
+[nosetests]
+where=../../test/rhsmlib_test
```
Comment (jirihnidek) [inline packages/rhsm/setup.cfg]: Directory: `../../test/rhsmlib_test` contains unit tests for `rhsmlib` module. Unit tests for `rhsm` module are in `../../test/rhsm/unit`.

File: packages/rhsm/README.md [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+1) From this directory, run `pipenv --three` This creates a new virtual env for just this package
+1) Run the following to install project deps in the virtualenv: `pipenv install`
+1) Run the following to enter the virtual env: `pipenv shell`
+
+
+From inside this virtual env you can freely install python-rhsm and test it as you like.
+Just run `python ./setup.py install` to install and `python ./setup.py test` to run the tests.
+
+The source for this subpackage is located (from checkout root) at ./src/rhsm
+The tests for this subpackage is located (from checkout root) at ./test/rhsmlib_test
```
Comment (jirihnidek) [inline packages/rhsm/README.md]: Unit tests are located at `./test/rhsm/unit`

File: packages/rhsm/setup.cfg (hunk 1 above)
Comment (jirihnidek) [reply packages/rhsm/setup.cfg]: Anyway, I wasn't able to run test for rhsm.

File: packages/rhsm/setup.py [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    url="http://www.candlepinproject.org",
+    description="Manage Red Hat Subscriptions",
+    license="GPLv2",
+    author="Chris Snyder",
+    author_email="chainsaw@redhat.com",
+    packages=find_packages('../../src', include=["rhsm"]),
+    package_dir={
+        "rhsm": "../../src/rhsm"
+    },
+    tests_require=test_require,
```
Comment (jirihnidek) [inline packages/rhsm/setup.py]: It is necessary to build binary module `certificate.so`. Look at `./setup.py` of subscription-manager.

File: packages/rhsm/setup.py [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    'nose'
+]
+
+setup(
+    name="python-rhsm",
+    version="1.22.0",
+    url="http://www.candlepinproject.org",
+    description="Manage Red Hat Subscriptions",
+    license="GPLv2",
+    author="Chris Snyder",
```
Comment (jirihnidek) [inline packages/rhsm/setup.py]: No, you will be honored to be the author. ;-)

File: packages/rhsm/README.md (hunk 2 above)
Comment (jirihnidek) [reply packages/rhsm/README.md]: Done

File: packages/rhsm/setup.py (hunk 3 above)
Comment (jirihnidek) [reply packages/rhsm/setup.py]: Done

**Comment by kahowell** [comment]
Resolves #1791

**Comment by kahowell** [comment]
Was not able to use `pip install .` from the subdirectory. Not sure why...

**Comment by jirihnidek** [comment]
@kahowell did you run `pipenv shell` before running `pip install .`? I'm asking, because it works for me.

> Was not able to use `pip install .` from the subdirectory. Not sure why...

**Comment by kahowell** [comment]
>@kahowell did you run pipenv shell before running pip install .? I'm asking, because it works for me. You can find more information about pipenv in README.md.

Nope still no go for me. I'm open to merging the PR anyways, as long as pip install git+https works... trying that now.

**Comment by kahowell** [comment]
Hrmmm... I tried `pip install 'git+https://github.com/candlepin/subscription-manager.git@2dada7fedd3857a56bbcc18154f90e89b14812ab#egg=rhsm&subdirectory=packages/rhsm'` but buried in the error was: `can't create build/temp.linux-x86_64-3.6/../../src/certificate.o: No such file or directory` . This was after using `pipenv shell` ... So I think maybe additional improvements could be made wrt the native dependency... Not sure what is necessary though.

**Comment by kahowell** [comment]
@vritant @jirihnidek thanks for all the work put into this, but after finding out that it is perfectly acceptable (though admittedly a bit confusing) for a distribution package to contain more than one import package (setuptools does this; see https://github.com/pypa/python-packaging-user-guide/issues/199#issuecomment-162746260), I decided that it is best to just have people use subscription-manager as the dependency, and let us work through any issues there.

**Pull request closed without merging**